gateway/src/apicast/balancer.lua

```diff
@@ -1,13 +1,22 @@
 local round_robin = require 'resty.balancer.round_robin'
+local resty_url = require 'resty.url'
+local empty = {}
 
 local _M = { default_balancer = round_robin.new() }
 
+local function get_default_port(upstream_url)
+  local url = resty_url.split(upstream_url) or empty
+  local scheme = url[1] or 'http'
+  return resty_url.default_port(scheme)
+end
+
 function _M.call(_, _, balancer)
   balancer = balancer or _M.default_balancer
   local host = ngx.var.proxy_host -- NYI: return to lower frame
   local peers = balancer:peers(ngx.ctx[host])
 
-  local peer, err = balancer:set_peer(peers)
+  local port = get_default_port(ngx.var.proxy_pass)
```

Comment:
I checked the nginx docs and looks like there is no built in variable to get scheme of the upstream server. Pity. 
We should evaluate this only when the port is missing.